docs/src/main/asciidoc/mongodb-panache.adoc

```diff
@@ -1238,3 +1238,285 @@ by the presence of the marker file `META-INF/panache-archive.marker`. Panache in
+
+[source,java]
+----
+import javax.enterprise.context.RequestScoped;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.mongodb.panache.common.MongoDatabaseResolver;
+import io.vertx.ext.web.RoutingContext;
+
+@Dependent
+public class MongoConfig {
+ 
+    @Unremovable
+    @RequestScoped
+    public MongoDatabaseResolver mongoDatabaseResolver(RoutingContext context) {
+        return new CustomMongoDatabaseResolver(context);
+    }
+    
+}
+---- 
+====
+
+[NOTE]
+====
+If you also use xref:security-openid-connect-multitenancy.adoc[OIDC multitenancy], then if the OIDC tenantID and MongoDB 
+database are the same and must be extracted from the Vert.x `RoutingContext` and then you can pass the tenant id from the OIDC `TenantResolver` 
+to the MongoDB with Panache `MongoDatabaseResolver` as a `RoutingContext` attribute, for example:
+
+[source,java]
+----
+import io.quarkus.mongodb.panache.common.MongoDatabaseResolver;
+import io.vertx.ext.web.RoutingContext;
+
+@Unremovable
+@RequestScoped
+public class CustomMongoDatabaseResolver implements MongoDatabaseResolver {
+
+    @Inject
+    RoutingContext context;
+    ...
+    @Override
+    public String resolve() {
+        // OIDC TenantResolver has already calculated the tenant id and saved it as a RoutingContext `tenantId` attribute:
+        return context.get("tenantId");
+    }
+}
+----
+====
+
+Given this entity:
+
+[source,java]
+----
+import org.bson.codecs.pojo.annotations.BsonId;
+import io.quarkus.mongodb.panache.common.MongoEntity;
+
+@MongoEntity(collection = "persons")
+public class Person {
+    @BsonId
+    public Long id;
+    public String firstName;
+    public String lastName;
+}
+----
+
+And this repository:
+
+[source,java]
+----
+import javax.enterprise.context.ApplicationScoped;
+import io.quarkus.mongodb.panache.PanacheMongoRepositoryBase;
+
+@ApplicationScoped
+public class PersonRepository implements PanacheMongoRepositoryBase<Person, Long> {
+   //...
+}
+----
+
+And this resource:
+[source,java]
+----
+import java.net.URI;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+@Path("/persons")
+public class PersonResource {
+
+    @Inject
+    PersonRepository personRepository;
+
+    @GET
+    @Path("/{id}")
+    public Person get(Long id) {
+        return personRepository.findById(id);
+    }
+
+    @POST
+    public Response create(Person person) {
+        personRepository.persist(person);
+        return Response.created(URI.create(String.format("/persons/%d", person.id))).build();
+    }
+}
+----
+
+
+=== Configuring the application
+
+The same mongo connection will be used for all tenants, so a database has to be created for every tenant.
+
+[source,properties]
+----
+quarkus.mongodb.connection-string=mongodb://login:pass@mongo:27017
+# The default database
+quarkus.mongodb.database=sanjoka
+----
```

Comment:
Is it relly needed ? As the same connection will be used and the database config property will not be used I think this can be removed.

Reply:
if the resolver returns null, the default database will be used:
![image](https://user-images.githubusercontent.com/47221187/210802286-01ecead8-0f77-4d51-af96-e8cca5f18215.png)

Reply:
Yes, so when using multitenancy this is optional in case we didn't find the tenant.